modules/standard/Regex.chpl

```diff
@@ -528,14 +528,43 @@ proc compile(pattern: ?t, posix=false, literal=false, noCapture=false,
       if m then do_something_if_matched();
       if !m then do_something_if_not_matched();
 
+    .. warning::
+      offset field is deprecated, use byte offset instead
+    .. warning::
+      size field is deprecated, use numBytes instead
  */
 record regexMatch {
   /* true if the regular expression search matched successfully */
   var matched:bool;
   /* 0-based offset into the string or channel that matched; -1 if matched=false */
-  var offset:byteIndex; // 0-based, -1 if matched==false
+  var byteOffset:byteIndex;
   /* the length of the match. 0 if matched==false */
-  var size:int; // 0 if matched==false
+  var numBytes:int;
+
+  //TODO: remove when removing offset and size
```

Comment:
I think these initializers aren't necessary any more, since offset and size are not fields (and so no longer impacting the generated initializer)